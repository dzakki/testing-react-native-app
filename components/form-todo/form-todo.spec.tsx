import { act, render, screen, userEvent } from '@testing-library/react-native';
import FormTodo from './index';

describe('FormTodo', () => {
  const props = {
    onSubmit: jest.fn(),
  };

  const inputPlaceholder = 'write your todo here...';

  test('should render content correctly', () => {
    render(<FormTodo {...props} />);
    expect(screen.getByText('Submit')).toBeVisible();
    expect(screen.getByPlaceholderText(inputPlaceholder)).toBeVisible();
  });

  test('submit and call onSubmit function', async () => {
    const todoTitle = 'my todo';
    const event = userEvent.setup();
    jest.useFakeTimers();
    render(<FormTodo {...props} />);

    const inputTodo = screen.getByPlaceholderText(inputPlaceholder);
    await event.type(inputTodo, todoTitle);
    expect(inputTodo).toHaveProp('value', todoTitle);
    const btnSubmit = screen.getByText('Submit');
    await event.press(btnSubmit);
    expect(props.onSubmit).toHaveBeenCalledWith(todoTitle);
    jest.useRealTimers();
  });

  test('should match snapshot', () => {
    render(<FormTodo {...props} />);

    expect(screen.toJSON()).toMatchSnapshot();
  });
});

