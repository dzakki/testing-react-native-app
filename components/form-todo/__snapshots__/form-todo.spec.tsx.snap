// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormTodo should match snapshot 1`] = `
<View
  style={
    {
      "flexDirection": "column",
      "flexWrap": "wrap",
      "marginBottom": 8,
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    placeholder="write your todo here..."
    style={
      {
        "borderColor": "black",
        "borderRadius": 8,
        "borderWidth": 1,
        "marginBottom": 8,
        "padding": 8,
      }
    }
    value=""
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#DDDDDD",
        "borderRadius": 8,
        "marginBottom": 8,
        "opacity": 1,
        "padding": 8,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "black",
          "textAlign": "center",
        }
      }
    >
      Submit
    </Text>
  </View>
</View>
`;
